dags/analyzer_helper/discord/discord_transform_raw_data.py

```diff
@@ -1,59 +1,114 @@
 from datetime import datetime
-
-from dags.analyzer_helper.discord.transform_raw_data_base import TransformRawDataBase
+from analyzer_helper.discord.transform_raw_data_base import TransformRawDataBase
+import logging
 
 
 class DiscordTransformRawData(TransformRawDataBase):
     def transform(self, raw_data: list, platform_id: str, period: datetime) -> list[dict]:
         transformed_data = []
         for data in raw_data:
-            interactions = []
-            if data.get("replied"):
-                interactions.append({
-                    "name": "reply",
-                    "users_engaged_id": [data.get("replied")],
-                    "type": "receiver"
-                })
-
-            if data.get("user_mentions"):
-                interactions.append({
-                    "name": "mention",
-                    "users_engaged_id": data.get("user_mentions"),
-                    "type": "emitter"
-                })
-
-            if data.get("reactions"):
-                for reaction in data.get("reactions"):
-                    parts = reaction.split(", ")
-                    if len(parts) > 2:
+            try:
+                if "author" not in data or not data["author"]:
+                    raise ValueError("Missing 'author' in raw data")
+
+                interactions = []
+
+                if data.get("replied_user"):
+                    interactions.append({
+                        "name": "reply",
+                        "users_engaged_id": [data["replied_user"]],
+                        "type": "emitter"
+                    })
+                    # Create another document for the receiver
+                    receiver_interaction = {
+                        "author_id": data["replied_user"],
+                        "date": period,
+                        "source_id": data["messageId"],
+                        "metadata": {
+                            "thread_id": data["threadId"],
+                            "channel_id": data["channelId"],
+                            "bot_activity": data["isGeneratedByWebhook"] or data["botActivity"],
+                            # "channel_name": data["channelName"],
+                            # "thread_name": data["threadName"],
+                        },
+                        "actions": [
+                            {
+                                "name": "reply",
+                                "type": "receiver"
+                            }
+                        ],
+                        "interactions": []
+                    }
+                    transformed_data.append(receiver_interaction)
+
+                if data.get("user_mentions"):
+                    interactions.append({
```

Comment:
Please create another document for user_mentions as well. the author would be each user in user_mentions and there would be an interaction for it with name `mention`, type as `receiver`, and `users_engaged_id` with one id which was our primary document `author` (by primary document I meant the `data` variable which we're assigning value in the for loop)